test/features/sync.test.js

```diff
@@ -1,64 +1,695 @@
+const {initializeWithTab} = require("../common");
+
 describe("Sync", () => {
+  beforeEach(async () => {
+    this.webExt = await initializeWithTab();
+    this.syncHelper = new SyncTestHelper(this.webExt);
+  });
+
+  afterEach(() => {
+    this.webExt.destroy();
+    delete this.syncHelper;
+  });
+
+  it("testIdentityStateCleanup", async () => {
+    await this.syncHelper.stopSyncListeners();
+    console.log("Testing the cleanup of local storage");
+
+    await this.syncHelper.setState({}, LOCAL_DATA, TEST_CONTAINERS, []);
+
+    await this.webExt.browser.storage.local.set({
+      "identitiesState@@_firefox-container-5": { 
+        "hiddenTabs": [] 
+      }
+    });
+    console.log("local storage set: ", await this.webExt.browser.storage.local.get());
+
+    await this.webExt.background.window.identityState.storageArea.upgradeData();
+
+    const macConfigs = await this.webExt.browser.storage.local.get();
+    const identities = [];
+    for(const configKey of Object.keys(macConfigs)) {
+      if (configKey.includes("identitiesState@@_") && !configKey.includes("default")) {
+        identities.push(macConfigs[configKey]);
+      }
+    }
+
+    console.assert(identities.length === 5, "There should be 5 identity entries");
```

Comment:
Here an example how asserting with `should` could look like:

```suggestion
    identities.should.have.lengthOf(5);
    for (const identity of identities) {
      identity.macAddonUUID.should.be.ok;
    }
```

[chai reference](https://www.chaijs.com/api/bdd/)